tests/test_scheduler.py

```diff
@@ -249,6 +274,17 @@ def test_winhpc(self):
         self.assertEqual(fluentOpts, "-t32 -cnf=M0:8,M1:8,M2:16")
         del os.environ["CCP_NODES"]
 
+    def test_slurm_single_num(self):
+        os.environ["SLURM_JOB_NODELIST"] = "M[1-2],M[3]"
```

Comment:
Interesting, was this use case not working before?  Or, was it working but returned the wrong result?

Reply:
There was not a test built for this case. It was in load_machine.py. I did have to make a small change in the regex (lines 336 and 375) to get the correct result.